src/amplitude-client.js

```diff
@@ -225,6 +227,34 @@ AmplitudeClient.prototype.init = function init(apiKey, opt_userId, opt_config, o
   }
 };
 
+/**
+ * @private
+ */
+AmplitudeClient.prototype._migrateUnsentEvents = function _migrateUnsentEvents(cb) {
```

Comment:
I'm curious why we have this callback chaining pattern in `init`? Is it because we need it to interact with AsyncStorage? in the case where `init` is just a consecutive / synchronous you could just write the logic all straight out without any callbacks.

Reply:
to make sure they're not touching asyncStorage at the same time, e.g. making it synchronous via migrating unsent events before accessing it